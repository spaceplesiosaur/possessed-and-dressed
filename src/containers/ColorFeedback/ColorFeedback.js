import React from 'react';
import { Link } from 'react-router-dom'
import './ColorFeedback.scss'
import { connect } from 'react-redux'

export const ColorFeedback = ({match, chosenColor, chosenHost, season}) => {

  const determineFeedback = () => {
    return (!match)
      ? (<p className="colorFeedback-main-feedback">Terrible choice! Your host's season is {`${season.toUpperCase()}`}. {`${chosenColor.name.toUpperCase()}`} will make them look tired, sad and afraid, and the church will surely try to perform an exorcism on them!</p>)
      : (<p className="colorFeedback-main-feedback">Great choice! Your host's season is {`${season.toUpperCase()}`}. {`${chosenColor.name.toUpperCase()}`} will look great on them - nobody will suspect a thing!  Happy haunting!</p>)
  }
  return (
    chosenColor.name && <section className="colorFeedback-main">
      <div className="colorFeedback-main-colorBox" style={{backgroundColor: `${chosenColor.hex_code}`}}>
      </div>
      <h2 className="colorFeedback-main-name">{chosenColor.name.toUpperCase()}</h2>
      {determineFeedback()}
      <button className="back-button" onClick={()=> window.location = '/'}>Back</button>
    </section>
  )
}

export const mapStateToProps = (state) => ({
  chosenColor: state.chosenColor,
  chosenHost: state.chosenHost
})
export default connect(mapStateToProps, null)(ColorFeedback)
