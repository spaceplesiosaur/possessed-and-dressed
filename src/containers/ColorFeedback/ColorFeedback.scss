.colorFeedback-main-colorBox {
  display: flex;
  flex-direction: column;
  height: 200px;
  width: 200px;
}
