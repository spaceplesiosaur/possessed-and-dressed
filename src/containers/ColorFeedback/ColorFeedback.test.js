import React from 'react';
import { ColorFeedback } from './ColorFeedback';
import { shallow } from 'enzyme';

describe('ColorFeedback', () => {
  let wrapper;
  let mockChosenColor;
  let mockChosenHost;
  let mockHostSeason

  beforeEach(() => {
    mockChosenColor = {
        id: 35,
        name: "neon strawberry",
        hex_code: "#F6003C",
        category: 34
    }

    mockChosenHost = {
        id: 1,
        name: "Blucifer",
        picture: "https://i.imgur.com/I5MqMUV.jpg",
        season: 9
    }

    mockHostSeason = {
        id: 9,
        name: "winter",
        colors: [
            34,
            35,
            36,
            52,
        ]
    }

    wrapper = shallow(<ColorFeedback
      match={true}
      chosenColor={mockChosenColor}
      chosenHost={mockChosenHost}
      season={mockHostSeason}
      />)
  })

  it('should render with correct data when match is true', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with correct data when match is false', () => {
    const wrapper = shallow(<ColorFeedback
      match={false}
      chosenColor={mockChosenColor}
      chosenHost={mockChosenHost}
      season={mockHostSeason}
      />)
    expect(wrapper).toMatchSnapshot()
  })
})
