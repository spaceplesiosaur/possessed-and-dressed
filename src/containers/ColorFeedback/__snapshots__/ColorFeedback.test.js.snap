// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorFeedback should render with correct data when match is false 1`] = `
<section
  className="colorFeedback-main"
>
  <div
    className="colorFeedback-main-colorBox"
    style={
      Object {
        "backgroundColor": "#F6003C",
      }
    }
  />
  <h2
    className="colorFeedback-main-name"
  >
    neon strawberry
  </h2>
  <p>
    No! Your host's season is 
    [object Object]
    . The color 
    neon strawberry
     will make them look tired, sad and afraid, and someone will try to perform an exorcism on them!
  </p>
</section>
`;

exports[`ColorFeedback should render with correct data when match is true 1`] = `
<section
  className="colorFeedback-main"
>
  <div
    className="colorFeedback-main-colorBox"
    style={
      Object {
        "backgroundColor": "#F6003C",
      }
    }
  />
  <h2
    className="colorFeedback-main-name"
  >
    neon strawberry
  </h2>
  <p>
    Yes! Your host's season is 
    [object Object]
    . The color 
    neon strawberry
     will look great on them - nobody will suspect a thing!
  </p>
</section>
`;
